### PR TITLE
Support Top sql

### PR DESCRIPTION
**Pull request by wu-sheng**
I am trying to merge this to make sure the master ci works. The old master codes base on wrong query protocol commitid, the ci couldn't pass. See #2238 

Most of the top sql codes provided, just haven't implemented the query. @peng-yongsheng If the codes look like, please merge this first.

### CODE REVIEW

**Comment by wu-sheng** [comment]
I did a simple test in es storage.
- Query
```
{
  getTopNRecords(condition:{serviceId:6, 
    metricName:"top_n_database_statement",
    topN:30, 
    order:DES, 
    duration:{step:MINUTE, start: "2019-02-12 0100", end: "2019-02-12 2300"}
  }){
    statement 
    latency
    traceId
  }
}
```

- Result
```
{
  "data": {
    "getTopNRecords": [
      {
        "statement": "select * from database where complex = 1;",
        "latency": 950,
        "traceId": "127.128.129"
      },
      {
        "statement": "select * from database where complex = 1;",
        "latency": 950,
        "traceId": "139.140.141"
      },
      {
        "statement": "select * from database where complex = 1;",
        "latency": 950,
        "traceId": "163.164.165"
      },
      {
        "statement": "select * from database where complex = 1;",
        "latency": 950,
        "traceId": "175.176.177"
      },
      {
        "statement": "select * from database where complex = 1;",
        "latency": 950,
        "traceId": "211.212.213"
      }
}
```

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/data/LimitedSizeDataCollection.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        for (int i = 0; i < storageDataList.size(); i++) {
+            STORAGE_DATA storageData = storageDataList.get(i);
+            if (value.compareTo(storageData) <= 0) {
+                if (i == 0) {
+                    // input value is less than the smallest in top N list, ignore
+                } else {
+                    // Remove the smallest in top N list
+                    // add the current value into the right position
+                    storageDataList.removeFirst();
+                    storageDataList.add(i, value);
```
Comment (peng-yongsheng) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/data/LimitedSizeDataCollection.java]: add(I-1, value);

Init: 1,3,5,7,9
limit size: 5
income: 4
result: 3,5,4,7,9

**Comment by wu-sheng** [comment]
I just finished the H2 and MySQL storage tests. Passed.

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/manual/database/TopNDatabaseStatement.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+/**
+ * Database TopN statement, including Database SQL statement, mongoDB and Redis commands.
+ *
+ * @author wusheng
+ */
+@TopNType
+@StorageEntity(name = TopNDatabaseStatement.INDEX_NAME, builder = TopNDatabaseStatement.Builder.class, source = Scope.DatabaseSlowStatement)
+public class TopNDatabaseStatement extends TopN {
+    public static final String INDEX_NAME = "top_n_database_statement";
```
Comment (peng-yongsheng) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/manual/database/TopNDatabaseStatement.java]: index name looks like too long, append the namespace will cause the model to exceed database limits.

Comment (wu-sheng) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/manual/database/TopNDatabaseStatement.java]: `top_n_database_statement` is shorter than old existing one, such as `service_relation_client_cpm`

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/data/LimitedSizeDataCollection.java (hunk 1 above)
Comment (wu-sheng) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/data/LimitedSizeDataCollection.java]: Yes, this is not right. Fixed.

**Comment by wu-sheng** [comment]
@TinyAllen This is going to be merged soon, you could work on UI about slow db access statement list. If you have any issue, ping me.

**Comment by wu-sheng** [comment]
@JaredTan95 @liuhaoyang The backend is ready, welcome to review, recheck and test. UI should be ready soon.

**Pull request merged**